junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedTestParameterResolver.java

```diff
@@ -72,61 +63,14 @@ public boolean supportsParameter(ParameterContext parameterContext, ExtensionCon
 		}
 
 		// Else fallback to behavior for parameterized test methods without aggregators.
-		return parameterIndex < this.arguments.length;
+		return parameterIndex < this.arguments.getConsumedLength();
 	}
 
 	@Override
 	public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext)
 			throws ParameterResolutionException {
-		return this.methodContext.resolve(parameterContext, extensionContext, extractPayloads(this.arguments),
+		return this.methodContext.resolve(parameterContext, extensionContext, this.arguments.getConsumedPayloads(),
 			this.invocationIndex);
 	}
 
-	/**
-	 * @since 5.8
-	 */
-	@Override
-	public void afterTestExecution(ExtensionContext context) {
```

Comment:
:speech_balloon: This was not called when, for example, a `@BeforeEach` method failed. The registration now happens earlier in `ParameterizedTestInvocationContext.prepareInvocation` to ensure it's executed in those cases as well.